### DISCUSSION TITLE
Extend integration to support two heating circuit and solar thermal

### DISCUSSION THREAD

**Discussion by dominikamann** [dominikamann/oekofen-pellematic-compact#3] [Ideas]
@lucaferrucci:

hello, good morning, as promised I am sending you the file of my system, it is quite complete as I have 2 heating circuits with mixer and solar thermal, if you need anything else, ask without problems.
I have excellent relations with oekofen italia and they really liked your work

[jasonAll.zip](https://github.com/dominikamann/oekofen-pellematic-compact/files/10498657/jasonAll.zip)

**Comment by dominikamann** [comment]
@lucaferrucci i implemented the second heating circuit, thermal and solar gain support. Please check with your heater! Release v0.0.5.

**Reply by lucaferrucci** [reply]
YESS!!!!
it's all ok
Thank you veri much!!!!!!!

**Reply by dominikamann** [reply]
Iam happy to hear that. If you need further support or features let me know ;)

**Comment by lucaferrucci** [comment]
The same goes for you on the okofen system, what software version do you have on the pellematic?






https://software.oekofen.info/my_files/ <https://software.oekofen.info/my_files/>

usr and psx :software

**Reply by dominikamann** [reply]
When I check the firmeware in the app, it tells me it is "Touch V4.00b". The heater is fairly new. I had several problems with this version and reported them to customer service. They told me that they will fix the problem and release a new version in Q1 2023. My problem is that the software update cannot be installed automatically via the cloud / FOTA. They told me that this is only possible locally via an authorised dealer. This is a bit annoying, because my heating installer is overworked and doesn't know anything about "software" at all ... . They were overwhelmed with my questions ...

E.g. Pellet consumption issue:

Whenever the suction pump starts (6 pm), the pellet consumption daily jumps. A short time later it returns to normal. This is a big issue when you want to calculate the daily pellet usage as difference. E.g. in Month December you needed 100kg pellets. With this bug this is not possible :/. The customer support told me that the consumption values are wrong overall or at least they have problems with them (wrong values).

![grafik](https://user-images.githubusercontent.com/29973737/214562835-821a45d9-8a70-4fbd-bedc-be73e1012048.png)

**Comment by lucaferrucci** [comment]
absolutely, 4.0 has many problems, you can choose between 4.01 and 4.02, 4.01 solves the problems of 4 but does not measure the level of the main pellet tank well, in 4.02 it measures the pellet level well but "forgets" the consumption of the previous day.
From the link I sent you you can download all versions.
What tank do you have? 180 kg with hand loading or flexilo?

For pellet consumption, version 4.02 waits for the intermediate tank to be empty and then activates the aspirator, thus managing to calculate the subtraction well.
Do you know the local update procedure? I can give you all the info if you want

**Comment by lucaferrucci** [comment]
release note
[known-issues-stand-20122022-1.pdf](https://github.com/dominikamann/oekofen-pellematic-compact/files/10499876/known-issues-stand-20122022-1.pdf)

**Comment by dominikamann** [comment]
Thank you for the detailed informations and known issues 🙏 That helps a lot. 

It is a PES 212 (4-12 kW) / P0060B5_47657E with a Flexilo Compact. 

I would be very happy if you share the update guide/procedure with me 😊👍

**Comment by lucaferrucci** [comment]
you shouldn't lose the system settings, but insert a blank usb stick and save the settings, okofen recommends setting from "0".
do you have code to enter the advanced menu?
After you have saved the settings, if you have the thermostat with the display inside the house, you must first update it, then the pallamatic.
Disconnect the bus cable behind the display of the internal thermostat insert the usb key and switch on, it will do the update, after the same thing on the pellematic, switch off insert the key to update the whole system, if it keeps the settings it is ok, otherwise insert the key where you had saved the settings and loads everything if it goes well, otherwise you have to set up the whole system.
let me know if you have difficulties, but I don't think so, from what you did in home assistant I don't think so :-))

**Reply by dominikamann** [reply]
Awesome! Thank you very much. It's good to know how it works 👍

**Comment by lucaferrucci** [comment]
yes
